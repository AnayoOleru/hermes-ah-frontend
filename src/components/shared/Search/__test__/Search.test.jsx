import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import SearchComponent from '../Search';

describe('Search component', () => {
  it('should match snapshot', () => {
    const wrapper = create(<SearchComponent />);
    expect(wrapper.toJSON).toMatchSnapshot();
  });
  it('test handleResultSelect function', () => {
    const wrapper = shallow(<SearchComponent />);
    const event = { preventDefault: jest.fn() };
    const result = { title: 'test' };
    expect(wrapper.instance().handleResultSelect(event, { result }));
  });

  it('test handleSearchChange function', async () => {
    const wrapper = shallow(<SearchComponent />);
    const event = { preventDefault: jest.fn() };
    const value = { title: 'test' };
    const newAuthors = [];
    const initialState = {
      isLoading: false,
      results: [],
      value: '',
      getResults: {
        articles: {},
        authors: {},
        tags: {},
      },
    };
    expect(newAuthors).toBeDefined();
    expect(await wrapper.instance().handleSearchChange(event, { value }));
    expect(
      (wrapper.instance().state = {
        initialState,
      })
    );
  });
});
