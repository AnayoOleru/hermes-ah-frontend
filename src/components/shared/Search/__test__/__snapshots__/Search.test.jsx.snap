// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `[Function]`;
